### ISSUE TITLE
Wrong Content-Type when calling token api

### ISSUE THREAD

**Issue by slashmili** [scrogson/oauth2#90]
I'm using oauth2 library with:

```
oauth2: 0.9.0
hackney: 1.6.6
auth server: google
```
```
elixir --version
Erlang/OTP 19 [erts-8.2] [source] [64-bit] [smp:8:8] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

Elixir 1.4.0
```

# Problem
When I try to get token with `OAuth2.Client.get_token` function I get failure from google OAuth server saying:
```
{"error" : "invalid_request"}
```
I debugged the request the we made from Elixir(hackney). it's like this:

```
accept: application/x-www-form-urlencoded
Content-Type: application/x-www-form-urlencoded, application/x-www-form-urlencoded, application/x-www-form-urlencoded, application/x-www-form-urlencoded
User-Agent: hackney/1.6.6
Host: 10.211.55.9:8000
Content-Length: 307

auth_scheme=request_body&client_id=XXXX&client_secret=SECCCC&code=CODE&grant_type=authorization_code&redirect_uri=https%3A%2F%2Flocalhost%3A4443%2F%2Fo%2Foauth%2Fcallback
```

And as you see `Content-Type` is somehow wrong.

I tried to call google oauth token with only one `application/x-www-form-urlencoded` and `application/x-www-form-urlencoded, application/x-www-form-urlencoded, application/x-www-form-urlencoded, application/x-www-form-urlencoded` and looks like google doesn't accept the latter as valid `Content-Type`

I even tried to change `ouath2` code and only send one `content-type`:
```
    headers = [{"content-type", "application/x-www-form-urlencoded"}]
    case :hackney.request(method, url, headers, body, req_opts) do
      {:ok, ref} when is_reference(ref) ->
```

However still hackney was sending two content type:
```
Content-Type: application/x-www-form-urlencoded, application/x-www-form-urlencoded
User-Agent: hackney/1.6.6
Host: 10.211.55.9:8000
Content-Length: 307
```

# Possible Fix
I tried to make a request using hackney `1.7.1` and [looks like they fixed](https://github.com/benoitc/hackney/commit/51c5dd845e69f80da148919189cc1fa85b52cdfa) it on that version.

Is it possible to use new version of `hackney` in `oauth2`?

**Comment by slashmili** [comment]
BTW I have to mention google changed API behaviour today, I've been using the same code past few days and it stops working since morning

**Comment by slashmili** [comment]
I also ran this code from both versions and you can see the result here https://requestb.in/1dlgnti1?inspect

```elixir
header =  [{"accept", "application/x-www-form-urlencoded"}, {"content-type", "application/x-www-form-urlencoded"}, {"content-type", "application/x-www-form-urlencoded"}]
:hackney.request(:post, "http://requestb.in/1dlgnti1", header, "request=data", [])
```

**Comment by scrogson** [comment]
Fixed in #91
